core/app/models/spree/refund.rb

```diff
@@ -5,10 +5,14 @@ class Refund < Spree::Base
     include Metadata
 
     belongs_to :payment, inverse_of: :refunds, optional: true
-    belongs_to :reason, class_name: 'Spree::RefundReason', foreign_key: :refund_reason_id, optional: true
+    belongs_to :reason,
+      class_name: 'Spree::RefundReason',
+      foreign_key: :refund_reason_id,
+      optional: true,
+      inverse_of: :refunds
     belongs_to :reimbursement, inverse_of: :refunds, optional: true
 
-    has_many :log_entries, as: :source
+    has_many :log_entries, as: :source, dependent: :destroy
```

Comment:
Not sure about this 🤔 . Actually I do not think refunds should ever be allowed to be destroyed. They are kind of legal records. I know that this just makes sure that *if* we delete a refund the log entry is removed as well, but I am not so sure about this as well.